crates/client/eth/src/l1_gas_price.rs

```diff
@@ -67,6 +68,21 @@ async fn update_gas_price(eth_client: &EthereumClient, l1_gas_provider: GasPrice
     l1_gas_provider.update_eth_l1_gas_price(*eth_gas_price);
     l1_gas_provider.update_eth_l1_data_gas_price(avg_blob_base_fee);
 
+    // fetch eth/strk price and update
+    if let Some(oracle_provider) = &l1_gas_provider.oracle_provider {
+        let (eth_strk_price, decimals) =
+            oracle_provider.fetch_eth_strk_price().await.context("failed to retrieve ETH/STRK price")?;
+        let strk_gas_price = (BigDecimal::new((*eth_gas_price).into(), decimals.into())
+            / BigDecimal::new(eth_strk_price.into(), decimals.into()))
+        .as_bigint_and_exponent();
+        let strk_data_gas_price = (BigDecimal::new(avg_blob_base_fee.into(), decimals.into())
+            / BigDecimal::new(eth_strk_price.into(), decimals.into()))
```

Comment:
Same issue here

Reply:
added the check here : crates/primitives/oracle/src/pragma.rs:59

when the api return a 0 it say that we failed to fetch the price for this specific reason

Reply:
Fair, but I still don't very like the fact that this function can be called with data that results in a division by 0. Imo a check should still be added here as it is still possible to introduce hard to detect crashes if this function were to be misused in the future.

Reply:
I'm not sure to get it. It can't have a division by 0 because the utils versions that retrieve the price return an error   if its zero.
if we want a to add other oracle in the future, the utils function that use it must check its own return data.